### PR TITLE
IA-2023: document filters in FE guidelines

### PR DESCRIPTION
**Pull request by quang-le**
Add some doc about setting up filters in iaso front-end

Related JIRA tickets : IA-2023

## Self proofreading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- Add doc about filters

## How to test

Just read the thing

### CODE REVIEW

**Review by beygorghor — APPROVED** [review]
❤️

**Comment by beygorghor** [comment]
just check black ...

File: docs/pages/dev/reference/front-end_guidelines/front-end_guidelines.md [hunk 1]
```diff
@@ -214,6 +214,41 @@ export const useDeleteWhatever = (
 };
 ```
 
+### Filters
+
+Most tables we display come with one or several filters, most commonly a text search and date filters.
+
+It's a global feature in Iaso that all filter searches performed on pages are deep-linked, i.e.: the parameters of the filters are saved in the url, so users can share the results of their search/filtering. This has an impact on the architecture of Iaso:
+
+- All search fields need to be declared in `routes.js`, under the `params` key. Important: `params` is an ordered list. Passing them in the wrong order in the url will resukt in a 404.
```
Comment (bmonjoie) [inline docs/pages/dev/reference/front-end_guidelines/front-end_guidelines.md]:
```suggestion
- All search fields need to be declared in `routes.js`, under the `params` key. Important: `params` is an ordered list. Passing them in the wrong order in the URL will result in a 404.
```

File: docs/pages/dev/reference/front-end_guidelines/front-end_guidelines.md [hunk 2]
```diff
@@ -214,6 +214,41 @@ export const useDeleteWhatever = (
...
+
+- Applying a filter doesn't change the component state per se but results in a redirection. Depending on the use cases, we may or may not want to save consecutive filters/searches in the routers history
+
+We have a few ready-made components for filters:
+
+- `InputComponent`: handles most types of inputs: text, select, checkbox, radio
+- `OrgUnitTreeviewModal`: handles searches on org units
+- `DatePicker` and `DateRangePicker`: handle dates
```
Comment (bmonjoie) [inline docs/pages/dev/reference/front-end_guidelines/front-end_guidelines.md]: AFAICT, `DatePicker` and `DateTimePicker` don't have a `keyValue` property.

Also, I couldn't find `DateRangePicker` in `bluesquare-components`:

https://github.com/BLSQ/bluesquare-components/tree/main/src/components

Comment (quang-le) [reply docs/pages/dev/reference/front-end_guidelines/front-end_guidelines.md]: Ah good one, I'll double-check

**Pull request merged**